src/parser/lval.js

```diff
@@ -34,8 +34,8 @@ pp.toAssignable = function (node, isBinding, contextDescription) {
         this.toAssignable(node.value, isBinding, contextDescription);
         break;
 
-      case "SpreadProperty":
-        node.type = "RestProperty";
+      case "SpreadElement":
```

Comment:
SpreadElement should only be converted to RestElement if isBinding === true and otherwise fail.

Previously we were not converting `SpreadElement` at all, but now we need to in some cases (the cases that were previously SpreadProperty)